src/zimdump.cpp

```diff
@@ -39,10 +40,32 @@
 # include <windows.h>
 #else
 # define SEPARATOR "/"
-# include <unistd.h>
-# include <sys/stat.h>
+#include <unistd.h>
 #endif
 
+#define ERRORSDIR "_errordir/"
+
+inline static void createdir(const std::string &path, const std::string &base)
+{
+    if (path.size() <= 1)
+        return ;
+
+    std::size_t position = 0;
+    while(position != std::string::npos)
+    {
+        position = path.find('/', position+1);
+        if (position != std::string::npos)
+        {
+            std::string fulldir = base + path.substr(0, position);
+            #if defined(_WIN32)
+              ::mkdir(fulldir.c_str());
```

Comment:
We use wide variants on windows to handle non ascii char in path.

Reply:
@mgautierfr I'm not sure on the necessity of using widechars in this case. The directory is taken from the arguments as a 8bit char array. We can have non ascii chars there.

Reply:
We already have a lot of issue with this on windows. It is far better to use widechar.

Reply:
Ok. I'll use widechars now and we can assess the issues on windows later.

If you agree I prefer to use a standard way of doing the conversion using stl's:
[wstring_convert](https://en.cppreference.com/w/cpp/locale/wstring_convert)

Reply:
It seems that `wstring_convert` is deprecated in C++17.

Reply:
yes because in C++17 we can use other stuff